src/server/pps/server/s3g_sidecar.go

```diff
@@ -376,14 +363,19 @@ func (h *handleJobsCtx) start() {
 			}
 
 			var key string
-			jobInfo := &pps.StoredJobInfo{}
+			jobInfo := &pps.JobInfo{}
 			if err := e.Unmarshal(&key, jobInfo); err != nil {
 				logrus.Errorf("sidecar s3 gateway watch unmarshal error: %v", err)
 			}
 
 			// create new ctx for this job
 			jobCtx, jobCancel := context.WithCancel(context.Background())
 			h.processJobEvent(jobCtx, e.Type, jobInfo.Job)
+			// TODO(2.0 required): this is not true - this will continue to get PUT
+			// notifications on every job update - this needs to be refactored
+			// (preferrably with just one watcher, will likely require a way to know
+			// when initial PUTs are complete so we can remove dead jobs after
+			// recovering from a failed watcher):
```

Comment:
Unrelated, but came across this in debugging - it looks like we're spawning a _lot_ of watches here (one every time the job is updated).  In addition, it doesn't look like the `.end` function ever returns?  Will file an issue tomorrow after talking to someone.

Reply:
@msteffen probably has the most context for this.